src/peft/peft_model.py

```diff
@@ -248,13 +255,43 @@ def save_pretrained(
                     # not supported in safetensors.
                     for shared_tensor_name in names[1:]:
                         output_state_dict[shared_tensor_name] = output_state_dict[shared_tensor_name].clone()
+                if convert_pissa_to_lora is not None:
+                    if not str(peft_config.init_lora_weights).startswith("pissa"):
+                        warnings.warn(
+                            "`convert_pissa_to_lora` only works for converting a PiSSA adapter to a LoRA adapter"
+                        )
+                    initial_adapter = os.path.basename(convert_pissa_to_lora)
+                    self.load_adapter(
+                        os.path.dirname(convert_pissa_to_lora), subfolder=initial_adapter, adapter_name=initial_adapter
+                    )
+                    if str(self.peft_config[initial_adapter].init_lora_weights).startswith("pissa"):
+                        raise ValueError(
+                            "The `init_lora_weights` parameter of the initial PiSSA adapter should be set to `True`."
+                            "Otherwise, `self.load_adapter` will subtract the principal singular value and vector again based on the residual model."
+                        )
+                    output_state_dict = self.base_model.subtract_pissa_init(output_state_dict, initial_adapter, kwargs)
 
                 safe_save_file(
                     output_state_dict,
                     os.path.join(output_dir, SAFETENSORS_WEIGHTS_NAME),
                     metadata={"format": "pt"},
                 )
             elif is_main_process:
+                if convert_pissa_to_lora is not None:
```

Comment:
Let's factor this out into a small sub-method so that we don't have a code duplication.